stl/inc/format

```diff
@@ -1496,6 +1496,218 @@ public:
     }
 };
 
+template <class _Ty>
+class _Fmt_buffer {
+private:
+    _Ty* _Ptr_        = nullptr;
+    size_t _Size_     = 0;
+    size_t _Capacity_ = 0;
+
+protected:
+    explicit _Fmt_buffer(const size_t _Size) noexcept : _Size_(_Size), _Capacity_(_Size) {}
+
+    ~_Fmt_buffer() = default;
+
+    _Fmt_buffer(_Ty* _Data, const size_t _Size, const size_t _Capacity) noexcept
+        : _Ptr_(_Data), _Size_(_Size), _Capacity_(_Capacity) {}
+
+    void _Set(_Ty* _Buf_data, const size_t _Buf_capacity) noexcept {
+        _Ptr_      = _Buf_data;
+        _Capacity_ = _Buf_capacity;
+    }
+
+    virtual void _Grow(size_t _Capacity) = 0;
+
+public:
+    using value_type = _Ty;
+
+    _Fmt_buffer(const _Fmt_buffer&) = delete;
+    void operator=(const _Fmt_buffer&) = delete;
+
+    _NODISCARD _Ty* begin() noexcept {
+        return _Ptr_;
+    }
+
+    _NODISCARD _Ty* end() noexcept {
+        return _Ptr_ + _Size_;
+    }
+
+    _NODISCARD size_t _Size() const noexcept {
+        return _Size_;
+    }
+
+    _NODISCARD size_t _Capacity() const noexcept {
+        return _Capacity_;
+    }
+
+    void _Clear() noexcept {
+        _Size_ = 0;
+    }
+
+    void _Try_resize(const size_t _Count) {
+        _Try_reserve(_Count);
+        _Size_ = _Count <= _Capacity_ ? _Count : _Capacity_;
+    }
+
+    void _Try_reserve(const size_t _New_capacity) {
+        if (_New_capacity > _Capacity_) {
+            _Grow(_New_capacity);
+        }
+    }
+
+    void push_back(const _Ty _Value) {
+        _Try_reserve(_Size_ + 1);
+        _Ptr_[_Size_++] = _Value;
+    }
+};
+
+struct _Fmt_buffer_traits {
+    explicit _Fmt_buffer_traits(ptrdiff_t) {}
+
+    _NODISCARD size_t _Count() const noexcept {
+        return 0;
+    }
+
+    _NODISCARD size_t _Limit(const size_t _Size) noexcept {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    ptrdiff_t _Count_ = 0;

```

Comment:
Urgh, this is backwards. Conventionally the members should be `_Count` and `_Limit` and the arguments passed `_Count_` and `_Limit_`

Reply:
Note that there are member functions `_Count` and `_Limit` in this class.